### DISCUSSION TITLE
Beside having the spock-report, generate the BDD .feature files

### DISCUSSION THREAD

**Discussion by AnthonyTonda** [spockframework/spock#1619] [Ideas]
.

**Comment by leonard84** [comment]
I have no idea what you are trying to ask.

**Reply by AnthonyTonda** [reply]
see the new post

**Reply by leonard84** [reply]
Why didn't you edit the question, instead of posting it as a reply?

**Comment by AnthonyTonda** [comment]
as Spock is a test framework that’s built on top of Groovy. It enforces BDD at the core and ensures all tests written using the framework follows the BDD conventions, it generates a report of the test execution and display the BDD conventions in the report.
Why not to have the option to generate the BDD feature in the file having the extension .feature.
A BDD .feature would be equivalent of the class.
the content of the feature file is having the BDD stories belong the case 

<img width="594" alt="Screenshot 2023-04-02 at 0 33 57" src="https://user-images.githubusercontent.com/109533717/229318935-592807ca-d124-44b3-b261-5570f880ae87.png">

**Reply by Vampire** [reply]
Are you talking about the `spock-reports` module that does not exist anymore since version 2.0?

**Reply by Vampire** [reply]
Or maybe about the 3rd party `spock-reports` extension that is not developed by the core project but by @renatoathaydes?

**Reply by leonard84** [reply]
Do you mean Cucumber's `.feature` file in Gherkin syntax?
If so, what would that accomplish? You can't execute Gherkin with Spock and vice-versa.

**Reply by AnthonyTonda** [reply]
@Vampire after checking the code of my teammate, I am speaking about Spock-reports extension.
@leonard84 yes I ,eat Cucumber's .feature file in Gherkin. I would be able to import that Cucumber .feature file to Xray. 
I am using the Spock-report to generate myself these Cucumber .feature by parsing it.
Would be better design that the Spock-report extension would do it.

**Reply by rtretyak** [reply]
Afaik spock-reports had (at least couple years ago) a pretty versatile templates system. Moreover, you can even feed it with your own 'ReportCreator' class to add even more customized behavior like I did [here](https://github.com/telstra/open-kilda/blob/develop/src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/TestCaseReportCreator.groovy) to get markdown reports. It should be relatively easy to achieve .feature-like reports with these tools 🤔 
By the way, spock-reports project is a separate project located here https://github.com/renatoathaydes/spock-reports

**Reply by AnthonyTonda** [reply]
Thank for all your time and your answers.
Will post in github 3rd party : Spock Reportd, for further clarification
